File: 백준/Bronze/2798. 블랙잭/블랙잭.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);

        int cardNum = scn.nextInt();
        int blackJackNum = scn.nextInt();

        int card[] = new int[cardNum];

        for (int i = 0; i < cardNum; i++){
            card[i] = scn.nextInt();
        }

        int myCard = 0;
        for (int first = 0; first < cardNum - 2; first++){
            for (int second = first + 1; second < cardNum - 1; second++){
                for (int third = second + 1; third < cardNum; third++){
                    int sum = card[first] + card[second] + card[third];
                    if (sum <= blackJackNum) myCard = Math.max(sum, myCard);
                }
            }
        }

        System.out.println(myCard);
    }
}

File: 백준/Bronze/2798. 블랙잭/블랙잭.cc
#include <iostream>
#include <cstdio>
#include <cstring>
#include <algorithm>
using namespace std;


int main()
{
    int N, M;
    cin >> N >> M;

    int card[N];
    memset(card, 0, sizeof(int));
    for (int i = 0; i < N; i++){
        cin >> card[i];
    }

    int sumOfThree = 0;

    for (int first = 0; first< N - 2; first++){
        for (int second = first + 1; second < N - 1; second++){
            for (int third = second + 1; third < N; third++){
                int temp = card[first] + card[second] + card[third];
                if (temp > sumOfThree && temp <= M){
                    sumOfThree = card[first] + card[second] + card[third];
                }
            }
        }
    }

    cout << sumOfThree;

 

}